crates/trycp_server/src/main.rs

```diff
@@ -271,6 +357,47 @@ fn os_eval(arbitrary_command: &str) -> String {
     }
 }
 
+fn send_json_rpc<S: Into<String>>(
+    uri: S,
+    method: &str,
+    params: serde_json::Value,
+) -> Result<serde_json::Value, String> {
+    let uri: String = uri.into();
+    let connection_uri = Url2::try_parse(uri.clone())
+        .map_err(|e| format!("unable to parse url:{} got error: {}", uri, e))?;
+    //        let config = WssConnectConfig::new(TlsConnectConfig::new(TcpConnectConfig::default()));
+    let config = WssConnectConfig::new(TcpConnectConfig::default());
+    let mut connection: InStreamWss<InStreamTcp> =
+        InStreamWss::connect(&connection_uri, config).map_err(|e| format!("{}", e))?;
+
+    connection
+        .write(
+            serde_json::to_vec(&JsonRpcRequest::new("1", method, params))
+                .unwrap()
+                .into(),
+        )
+        .unwrap();
+    connection.flush().map_err(|e| format!("{}", e))?;
+
+    let mut res = WsFrame::default();
+    loop {
```

Comment:
this could probably be replaced with a 

while let Err(e) =connection.read()
{
//process errors 
}

this would get rid of the loop break but not entirely necessary to refactor